modules/javafx.base/src/test/java/test/javafx/collections/ModifiableObservableListBaseTest.java

```diff
@@ -57,10 +60,47 @@ public ListIterator<String> listIterator() {
 
     @Nested
     class SetAllTest {
+        private final ObservableList<String> list = FXCollections.observableArrayList(List.of("a", "b", "c"));
+        private final List<String> recordedChanges = new ArrayList<>();
+
+        {
+            list.addListener((Change<? extends String> c) -> recordedChanges.add(c.toString()));
+        }
+
         @Test
         public void testNullArgumentThrowsNPE() {
-            var list = new MockModifiableObservableList(new ArrayList<>(List.of("a", "b", "c")));
             assertThrows(NullPointerException.class, () -> list.setAll((Collection<? extends String>) null));
+            assertThrows(NullPointerException.class, () -> list.replaceRange(0, 0, (Collection<? extends String>) null));
+            assertEquals(List.of(), recordedChanges);
+        }
+
+        @Test
+        void shouldRejectIllegalRange() {
+            assertThrows(IndexOutOfBoundsException.class, () -> list.replaceRange(-1, 2, List.of("d", "e", "f")));
+            assertThrows(IndexOutOfBoundsException.class, () -> list.replaceRange(0, 5, List.of("d", "e", "f")));
+            assertThrows(IndexOutOfBoundsException.class, () -> list.replaceRange(2, 1, List.of("d", "e", "f")));
+            assertEquals(List.of(), recordedChanges);
+        }
+
+        @Test
+        void shouldNotModifyListWhenOperationIsANoOp() {
+            assertFalse(list.replaceRange(1, 1, List.of()));
+            assertEquals(List.of("a", "b", "c"), list);
+            assertEquals(List.of(), recordedChanges);
+        }
+
+        @Test
+        void shouldReplaceElementsAtGivenRange() {
+            assertTrue(list.replaceRange(1, 3, List.of("B", "C", "D")));
```

Comment:
Would additional tests be useful?

* Replace a sub-range in the middle (so both beginning and end are kept)
* Replace a sub-range with fewer elements than are removed (the existing test checks the case of more elements added than removed)